crates/services/txpool_v2/src/pool.rs

```diff
@@ -224,30 +227,65 @@ where
     /// based on the constraints given in the configuration and the selection algorithm used.
     pub fn extract_transactions_for_block(
         &mut self,
-    ) -> Result<Vec<PoolTransaction>, Error> {
-        let extracted_transactions = self
-            .selection_algorithm
-            .gather_best_txs(
-                Constraints {
-                    max_gas: self.config.max_block_gas,
-                },
-                &mut self.storage,
-            )?
+        constraints: Constraints,
+    ) -> Vec<ArcPoolTx> {
+        self.selection_algorithm
+            .gather_best_txs(constraints, &mut self.storage)
             .into_iter()
             .map(|storage_entry| {
                 self.update_components_and_caches_on_removal(iter::once(&storage_entry));
 
                 storage_entry.transaction
             })
-            .collect::<Vec<_>>();
-
-        Ok(extracted_transactions)
+            .collect::<Vec<_>>()
     }
 
-    #[allow(dead_code)]
-    pub fn find_one(&self, tx_id: &TxId) -> Option<&PoolTransaction> {
+    pub fn find_one(&self, tx_id: &TxId) -> Option<ArcPoolTx> {
         Storage::get(&self.storage, self.tx_id_to_storage_id.get(tx_id)?)
-            .map(|data| &data.transaction)
+            .map(|data| data.transaction.clone())
+    }
+
+    pub fn contains(&self, tx_id: &TxId) -> bool {
+        self.tx_id_to_storage_id.contains_key(tx_id)
+    }
+
+    pub fn iter_tx_ids(&self) -> impl Iterator<Item = &TxId> {
+        self.tx_id_to_storage_id.keys()
+    }
+
+    /// Remove transaction but keep its dependents.
+    /// The dependents become executables.
+    pub fn remove_transaction(&mut self, tx_ids: Vec<TxId>) {
+        for tx_id in tx_ids {
+            if let Some(storage_id) = self.tx_id_to_storage_id.remove(&tx_id) {
```

Comment:
We are ok to try to delete the others if they are not already been removed yet. I'm not sure about the warning because this function can be called from a lot of differents place/modules around the code and even if it's suboptimal to call again this function (with an other module for example) on already deleted functions I don't think it's worth warning to the user about this has it will not make his node gone badly or break anything.

Reply:
If this doesn't break anything there's no need to log, indeed.

Thanks for explanation! :+1: